daemon/internal/nri/nri.go

```diff
@@ -82,10 +101,76 @@ func (n *NRI) Shutdown(ctx context.Context) {
 	n.nri = nil
 }
 
+// CreateContainer notifies plugins of a "creation" NRI-lifecycle event for a container,
+// allowing the plugin to adjust settings before the container is created.
+func (n *NRI) CreateContainer(ctx context.Context, ctr *container.Container) error {
+	n.mu.RLock()
+	defer n.mu.RUnlock()
+	if n.nri == nil {
```

Comment:
As a follow-up, we should look if we can come with more descriptive names for these; there's various places where we use `nri`  (both in package names and field-names), and it's not always clear "NRI-what? is it a bird, a plane, a service?"; e.g. also this one, which is a service that's started stopped, I _THINK_ ? https://github.com/moby/moby/blob/186a5abedccef0d36467cb559d83a60741885bc8/daemon/daemon.go#L120